hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/DatanodeLayout.java

```diff
@@ -96,11 +97,14 @@ public static void main(String[] args) {
     OzoneContainer.buildContainerSet(volumeSet, containerSet, conf);
     volumeSet.shutdown();
 
+    List<HddsVolume> failedVolumes = StorageVolumeUtil.getHddsVolumesList(
+        volumeSet.getFailedVolumesList());
+
     if (verify) {
-      for (HddsVolume vol : volumeSet.getFailedVolumesList()) {
+      for (HddsVolume vol : failedVolumes) {
         System.out.println("Failed Volume:" + vol.getHddsRootDir());
       }
     }
-    return volumeSet.getFailedVolumesList();
+    return failedVolumes;
```

Comment:
I think, its better to log if there is any failed metadata volume here as well.

Reply:
OK.

Reply:
Oh, here I think it is dedicated to HddsVolumes since this `dnlayout` command tries to verify the on-disk layout of all data paths or the single data path. So we don't have to check metadata volume here.